backend/src/main/java/botobo/core/quiz/application/QuizService.java

```diff
@@ -27,26 +24,33 @@ public QuizService(WorkbookRepository workbookRepository, CardRepository cardRep
         this.cardRepository = cardRepository;
     }
 
-    public List<QuizResponse> createQuiz(List<Long> ids) {
-        List<Card> cards = new ArrayList<>();
-        for (Long id : ids) {
-            final Workbook workbook = workbookRepository.findById(id)
-                    .orElseThrow(WorkbookNotFoundException::new);
-            cards.addAll(workbook.getCardsAsList());
+    public List<QuizResponse> createQuiz(List<Long> workbookIds) {
```

Comment:
```suggestion
    public List<QuizResponse> createQuiz(List<Long> workbookIds) {
        validateWorkbookIds(workbookIds);

        // 다 갖고와
        Quiz quiz = new Quiz(cardRepository.findCardsbyWorkbookIdIn(workbookIds));

        // 1. 또 보기 우선 선택하고 부족하면 또 보기가 아닌 카드 선택
        // 2. 선택된 놈들 shuffle
        // 3. 선택된 놈들 cancelNextQuiz
        Cards quizCards = quiz.selectQuizCards(10);

        return QuizResponse.cardsOf(quizCards);
    }
```

오즈의 리뷰를 보고, 그리고 오늘 리뷰를 남길 준비를 하면서 '또 보기' 여부와는 관계없이 모든 카드를 다 가져와도 좋겠다는 생각을 했어요. 코드도 간결해지고, 나중에 퀴즈에 포함될 카드 수가 10장보다 많아지면 결국 '또 보기' 카드 수보다 퀴즈에 필요한 카드 수가 더 많아질 것 같네요. 그리고 오즈가 말한 것 처럼 코드를 간단하게 만들고 다른 객체에 로직을 넣을 수 있다는 것도 좋아보임!! (번복 ㅈㅅ)

나름 Quiz 라는 객체를 만들어서 로직을 넣는 방식을 생각해봤어요! Cards 객체에 퀴즈 관련 로직을 넣기보다 새로 하나 만드는게 좋아보였음!

Reply:
Quiz 객체 생성하는거 좋은 것 같아요! 킹즈 의견이랑 기둥곰의견까지 절충해서 한번 리팩토링 해볼게요 ~~ 의견 남겨주셔서 감사해요 !!